smolyay/basis.py

```diff
@@ -608,6 +632,23 @@ def derivative(self, X, domain=None, n=1):
             y *= ((self.domain[1] - self.domain[0]) / (domain[1] - domain[0])) ** n
         return y
 
+    def integrate(self, domain=None):
+        """Integrate all basis functions
+        
+        Parameters
+        ----------
+        domain : numpy array of shape (2,)
+            the lower and upper bounds of X.
+            
+        Returns
+        -------
+        scalar or ndarray
+            the values of the basis functions."""
+        if domain is None:
+            return [x.integrate() for x in self.basis_functions]
+        else:
+            return [x.integrate()*(domain[1]-domain[0])/(self.domain[1]-self.domain[0]) for x in self.basis_functions]
```

Comment:
This method is supposed to return a NumPy array, but it doesn't currently. This would fit the pattern used above.
```suggestion
        result = numpy.array([x.integrate() for x in self.basis_functions])
        if domain is not None:
            result *= (domain[1]-domain[0])/(self.domain[1]-self.domain[0])
```